users/tests.py

```diff
@@ -1,3 +1,157 @@
+from django.utils import timezone
+# from datetime import timedelta, datetime
 
-# Create your tests here.
+
+LAST_LOGIN = timezone.now()
+CREATION_DATE = timezone.now()
+USERNAME = "testuser"
+FIRSTNAME = "Test"
+LASTNAME = "User"
+PASSWORD = "testpass"
+EMAIL = "testuser@gmail.com"
+PHONE_NUMBER = "0541234567"
+LONG_PROFESSION = " test test test test test test test test test test test test test test test test" \
+                 " test test test test test test test test test" \
+                 " test test test test test test test test" \
+                 "test_exceeding_max_length" * 50
+
+
+PROFIL_INFORMATION = {'username': ['testusername', 'testusername2', 'testusername3'],
+                      'password': ['testpassword', 'testpassword2', 'testpassword3'],
+                      'first_name': ['Bob', 'john', 'jane'],
+                      'last_name': ['Smith', 'Doe', 'Don'],
+                      'email': ['test@test.com', 'test2@test.com', 'test3@test.com'],
+                      'phone_number': ['0541234567', '0541234666', '0541234777'],
+                      'account_type': [Account_type.STUDENT, Account_type.STUDENT, Account_type.STUDENT],
+                      'city': ['New York', 'Tel Aviv', 'London'],
+                      'meeting_method': [Meeting_method.LIVE, Meeting_method.LIVE, Meeting_method.LIVE], }
+
+
+@pytest.fixture
+def make_profile(idx=0):
+    def make(
+        username: str = PROFIL_INFORMATION.get('username')[idx],
+        password: str = PROFIL_INFORMATION.get('password')[idx],
+        first_name: str = PROFIL_INFORMATION.get('first_name')[idx],
+        last_name: str = PROFIL_INFORMATION.get('last_name')[idx],
+        email: str = PROFIL_INFORMATION.get('email')[idx],
+        phone_number: str = PROFIL_INFORMATION.get('phone_number')[idx],
+        account_type: str = PROFIL_INFORMATION.get('account_type')[idx],
+        city: str = PROFIL_INFORMATION.get('city')[idx],
+        meeting_method: str = PROFIL_INFORMATION.get('meeting_method')[idx],
+    ):
+        profile = Profile.create(
+            username=username, password=password, account_type=account_type,
+            first_name=first_name, last_name=last_name, email=email,
+            phone_number=phone_number, city=city, meeting_method=meeting_method,
+        )
+        return profile
+
+    return make
+
+
+@pytest.fixture
+def new_profile():
+    return Profile.create(username=USERNAME, password=PASSWORD, account_type=Account_type.STUDENT,
+                          first_name=FIRSTNAME, last_name=LASTNAME)
+
+
+@pytest.mark.django_db()
+class TestUserProfileModel():
+
+    def test_profile(self, new_profile):
+        assert new_profile.user.username == USERNAME
+        assert new_profile.user.first_name == FIRSTNAME
+
+    def test_profile_creation(self, new_profile):
+        new_profile.save()
+
+    def test_profile_invalid(self, new_profile):
+        assert new_profile
+
+    # Test for invalid profession length
```

Comment:
if you feel there is a need for a comment, this means the test is not readable enough.

Reply:
deleted